spring-cloud-task-app-starters-docs/src/main/asciidoc/overview.adoc

```diff
@@ -153,3 +153,184 @@ public class TimestampTaskApplication {
 
 After doing so, you can simply add the additional configuration for the extra features of your application.
 
+=== Developing custom task applications and generating artifacts using the existing tooling
+
+In this section, we will explain how to develop a custom task application and then generate
+maven and docker artifacts for it using the existing tooling provided by the
+spring cloud task app starter infrastructure. For explanation purposes, we will assume that we are creating a new
+task application for a technology named foobar.
+
+* Create a new top level module named spring-cloud-starter-task-foobar
+
+Please follow the instructions above for designing a proper Spring Cloud Task starter. You may also look into the existing
+starters for how to structure a new one. Ensure that you name the main `@Configuration` class of your starter as
+`FoobarTaskConfiguration` as this is the default convention used by the app generation later. Please also make sure
+that the package for this `@Configuration` is `org.springfamework.cloud.task.app.foobar`. If you have a different
+class/package name, see below for how to override that in the app generator. The functionality name for which you create
```

Comment:
not clear what this overrding is about, seems confusing since the previous sentence says 'make sure that the package is..`.